_includes/guides/create-dockerfile.md

```diff
@@ -0,0 +1,39 @@
+A Dockerfile is a text document that contains the instructions to assemble a
+Docker image. When we tell Docker to build our image by executing the `docker build`
+command, Docker reads these instructions, executes them, and creates a Docker
+image as a result.
+
+Let’s walk through the process of creating a Dockerfile for our application. In
+the root of your project, create a file named `Dockerfile` and open this file in
+your text editor.
+
+> **What to name your Dockerfile?**
+>
+> The default filename to use for a Dockerfile is `Dockerfile` (without a file-
+> extension). Using the default name allows you to run the `docker build` command
+> without having to specify additional command flags.
+>
+> Some projects may need distinct Dockerfiles for specific purposes. A common
+> convention is to name these `Dockerfile.<something>` or `<something>.Dockerfile`.
+> Such Dockerfiles can then be used through the `--file` (or `-f` shorthand)
+> option on the `docker build` command. Refer to the
+> ["Specify a Dockerfile" section](/engine/reference/commandline/build/#specify-a-dockerfile--f)
+> in the `docker build` reference to learn about the `--file` option.
+>
+> We recommend using the default (`Dockerfile`) for your project's primary
+> Dockerfile, which is what we'll use for most examples in this guide.
+
+The first line to add to a Dockerfile is a [`# syntax` parser directive](/engine/reference/builder/#syntax).
+While _optional_, this directive instructs the Docker builder what syntax to use
+when parsing the Dockerfile, and allows older Docker versions with BuildKit enabled
+to upgrade the parser before starting the build. [Parser directives](/engine/reference/builder/#parser-directives)
+must appear before any other comment, whitespace, or Dockerfile instruction in
+your Dockerfile, and should be the first line in Dockerfiles.
+
+```dockerfile
+# syntax=docker/dockerfile:1
+```
+
+We recommend using `docker/dockerfile:1`, which always points to the latest release
```

Comment:
```suggestion
We recommend using `docker/dockerfile:1` as [Dockerfile frontend](https://github.com/moby/buildkit/blob/master/frontend/dockerfile/docs/syntax.md), which always points to the latest release
```

Reply:
Ideally, we would have the "docker file" reference unified. Currently part of it lives in `docker/cli`, and part of it (front-end syntax) lives in moby/buildkit.

That way users wouldn't have to head to GitHub to find the missing bits (but we may have to add additional styles for the docs to annotate options (a "badge"?) that explains they are buildkit-only, and require syntax X or up.